VSRAD.DebugServer/SharedUtils/ProcessUtils.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace VSRAD.DebugServer.SharedUtils
+{
+    public readonly struct ProcessTreeItem : IEquatable<ProcessTreeItem>
+    {
+        public int Id { get; }
+        public string Name { get; }
+        public int ChildLevel { get; }
+
+        public ProcessTreeItem(int id, string name, int childLevel)
+        {
+            Id = id;
+            Name = name;
+            ChildLevel = childLevel;
+        }
+
+        public bool Equals(ProcessTreeItem other) => Id == other.Id && Name == other.Name && ChildLevel == other.ChildLevel;
+        public override bool Equals(object obj) => obj is ProcessTreeItem data && Equals(data);
+        public override int GetHashCode() => Id;
+        public static bool operator ==(ProcessTreeItem left, ProcessTreeItem right) => left.Equals(right);
+        public static bool operator !=(ProcessTreeItem left, ProcessTreeItem right) => !(left == right);
+    }
+
+    public static class ProcessUtils
+    {
+        public static bool IsParentOf(this Process parentProcess, Process otherProcess)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                foreach (var line in File.ReadLines($"/proc/{otherProcess.Id}/status"))
+                {
+                    if (!line.StartsWith("PPid:", StringComparison.InvariantCultureIgnoreCase)) continue;
```

Comment:
Minor nitpick: according to [best practices](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings) it's more appropriate to use `OrdinalIgnoreCase` here. Relevant excerpts:
> * Use StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase for comparisons as your safe default for culture-agnostic string matching.
> * Use comparisons with StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase for better performance.
> * Do not use string operations based on StringComparison.InvariantCulture in most cases. One of the few exceptions is when you are persisting linguistically meaningful but culturally agnostic data.